### DISCUSSION TITLE
enforcing character limit while typing or copying and pasting text

### DISCUSSION THREAD

**Discussion by yuvaraj-thiagarajan** [bangle-io/bangle.dev#167] [Q&A] [answered]
Is there a way to enforce character limit while typing or pasting content from somewhere else? I am looking for an extension that trims longer text to a specified maximum length while copying and pasting content in addition to not allowing the users to type beyond a set character limit.

**Comment by kepta** [comment, marked as answer]
You will have to create a Prosemirror plugin which uses the [filterTransaction](https://prosemirror.net/docs/ref/#state.PluginSpec.filterTransaction) API.
